public/app/features/explore/AddToDashboard/index.tsx

```diff
@@ -43,39 +43,38 @@ interface Props {
 
 export const AddToDashboard = ({ exploreId }: Props) => {
   const [isOpen, setIsOpen] = useState(false);
-  const dispatch = useDispatch();
+  const { success } = useAppNotification();
   const selectExploreItem = getExploreItemSelector(exploreId);
 
   const { queries, panel, datasource } = useSelector((state: StoreState) => {
     const exploreItem = selectExploreItem(state);
     const queries = exploreItem?.queries || [];
     const datasource = exploreItem?.datasourceInstance;
-    const { graphFrames, logsFrames, nodeGraphFrames } = exploreItem?.queryResponse || {};
 
     return {
       queries,
       datasource: { type: datasource?.type, uid: datasource?.uid },
-      panel: getMainPanel(queries, graphFrames, logsFrames, nodeGraphFrames),
+      panel: getMainPanel(queries, exploreItem?.queryResponse),
     };
   });
 
-  const handleSave = async (data: SaveToNewDashboardDTO, redirect: boolean): Promise<void | ErrorResponse> => {
+  const handleSave = async (data: FormDTO, redirect: boolean): Promise<void | ErrorResponse> => {
     try {
-      const redirectURL = await addToDashboard(data, {
+      const { name, url } = await addToDashboard(data, {
         queries,
         datasource,
         panel,
       });
 
+      setIsOpen(false);
       if (redirect) {
-        locationService.push(redirectURL);
+        locationService.push(url);
```

Comment:
this needs to be changed to something like
```
const strippedUrl = locationUtil.stripBaseFromUrl(url);
locationService.push(strippedUrl);
```

to make it work in cases where grafana is served from sub path